### PR TITLE
Azure Storage Data Lake JVM support #2289

### PR DESCRIPTION
**Pull request by ppalaga**
Rel. #2289

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: extensions-jvm/azure-storage-datalake/integration-test/src/test/java/org/apache/camel/quarkus/component/azure/storage/datalake/it/AzureStorageDatalakeTest.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+            RestAssured.given()
+                    .delete("/azure-storage-datalake/filesystem/" + filesystem + "/path/" + filename)
+                    .then()
+                    .statusCode(204);
+
+            RestAssured.given()
+                    .delete("/azure-storage-datalake/filesystem/" + filesystem)
+                    .then()
+                    .statusCode(204);
```
Comment (aldettinger) [inline extensions-jvm/azure-storage-datalake/integration-test/src/test/java/org/apache/camel/quarkus/component/azure/storage/datalake/it/AzureStorageDatalakeTest.java:110]: I wonder if we should assert the deletions here.

**Review by aldettinger — APPROVED** [review]
Looks good to me.

**Review by jamesnetherton — APPROVED** [review]

File: extensions-jvm/azure-storage-datalake/integration-test/src/test/java/org/apache/camel/quarkus/component/azure/storage/datalake/it/AzureStorageDatalakeTest.java (hunk 1 above)
Comment (ppalaga) [reply extensions-jvm/azure-storage-datalake/integration-test/src/test/java/org/apache/camel/quarkus/component/azure/storage/datalake/it/AzureStorageDatalakeTest.java:110]: Yes, we could, although I wonder whether that level of paranoia wouldn't be better suited for the tests in Camel. I think here in Camel Quarkus the main purpose of the tests is to ensure that the main execution paths work flawlessly on Quarkus incl. native mode. If the delete request does not throw an exception, then I think the functionality can be considered proven. But yes, checking the state of the remote storage would be even safer.

Comment (aldettinger) [reply extensions-jvm/azure-storage-datalake/integration-test/src/test/java/org/apache/camel/quarkus/component/azure/storage/datalake/it/AzureStorageDatalakeTest.java:110]: Indeed, I agree it's not really the scope of camel-quarkus test.

**Comment by ppalaga** [comment]
JVM tests and azure passed - that should be enough

**Pull request merged**